### ISSUE TITLE
Support model "buffers"

### ISSUE THREAD

**Issue by Kaixhin** [ebetica/autogradpp#52]
Basically these are parameters that aren't updated via gradient descent (but would be serialized - a good example is the running mean or running variance in batch normalisation); they could also be completely fixed.

Simplest solution would be to just add Variables that don't need gradients within `initialize_parameters` and make sure optimizers skips anything with `requires_grad=false` (which should be done anyway).

**Comment by jgehring** [comment]
Note that you can serialize custom data in your model by implementing a custom `serialize` or `load/save` function.
